plugins/org.eclipse.sirius.table/src/org/eclipse/sirius/table/business/internal/refresh/DTableElementSynchronizer.java

```diff
@@ -781,10 +797,10 @@ private void doUpdateStyle(final DColumn column, final DTableElementStyle style)
         if (bestForegroundStyle != null) {
             boolean defaultStyleDescription = new StyleUpdaterQuery(styleUpdater).isDefaultForegroundColor(bestForegroundStyle.getForeGroundColor());
             colorUpdater.updateForegroundColor(style, bestForegroundStyle.getForeGroundColor(), defaultStyleDescription, column.getTarget());
-            if (bestForegroundStyle.getLabelFormat() != null && !Objects.equals(style.getLabelFormat(), bestForegroundStyle.getLabelFormat())) {
+            if (!Objects.equals(style.getLabelFormat(), bestForegroundStyle.getLabelFormat())) {
                 FontFormatHelper.setFontFormat(style.getLabelFormat(), bestForegroundStyle.getLabelFormat());
             }
-            if (bestForegroundStyle.getLabelSize() != -1 && style.getLabelSize() != bestForegroundStyle.getLabelSize()) {
+            if (style.getLabelSize() != bestForegroundStyle.getLabelSize()) {
```

Comment:
You remove "bestForegroundStyle.getLabelSize() != -1" but there is maybe a link with auto size. Have you checked ? Furthermore, you not remove it for the doUpdateStyle of a DLine.

Reply:
Handling labelSize  == -1 is performed by UI : org.eclipse.sirius.table.ui.tools.internal.editor.provider.DTableColumnLabelProvider
This is now also applied to doUpdateStyle(DLine, ...).